packages/material-ui/src/Tabs/Tabs.js

```diff
@@ -2,96 +2,214 @@ import * as React from 'react';
 import { isFragment } from 'react-is';
 import PropTypes from 'prop-types';
 import clsx from 'clsx';
-import { refType } from '@material-ui/utils';
+import { refType, deepmerge } from '@material-ui/utils';
+import { unstable_composeClasses as composeClasses } from '@material-ui/unstyled';
+import experimentalStyled from '../styles/experimentalStyled';
+import useThemeProps from '../styles/useThemeProps';
+import useTheme from '../styles/useTheme';
 import debounce from '../utils/debounce';
 import ownerWindow from '../utils/ownerWindow';
 import { getNormalizedScrollLeft, detectScrollType } from '../utils/scrollLeft';
 import animate from '../internal/animate';
 import ScrollbarSize from './ScrollbarSize';
-import withStyles from '../styles/withStyles';
-import TabIndicator from './TabIndicator';
 import TabScrollButton from '../TabScrollButton';
 import useEventCallback from '../utils/useEventCallback';
-import useTheme from '../styles/useTheme';
+import tabsClasses, { getTabsUtilityClass } from './tabsClasses';
 
-export const styles = (theme) => ({
-  /* Styles applied to the root element. */
-  root: {
-    overflow: 'hidden',
-    minHeight: 48,
-    WebkitOverflowScrolling: 'touch', // Add iOS momentum scrolling.
-    display: 'flex',
-  },
-  /* Styles applied to the root element if `orientation="vertical"`. */
-  vertical: {
-    flexDirection: 'column',
-  },
-  /* Styles applied to the flex container element. */
-  flexContainer: {
-    display: 'flex',
+const scrollbarSizeClassName = 'PrivateScrollbarSize';
+
+const useUtilityClasses = (styleProps) => {
+  const {
+    vertical,
+    fixed,
+    hideScrollbar,
+    scrollableX,
+    scrollableY,
+    flexContainerVertical,
+    centered,
+    scrollButtonsHideMobile,
+    classes,
+  } = styleProps;
+
+  const slots = {
+    root: ['root', vertical && 'vertical'],
+    scroller: [
+      'scroller',
+      fixed && 'fixed',
+      hideScrollbar && 'hideScrollbar',
+      scrollableX && 'scrollableX',
+      scrollableY && 'scrollableY',
+    ],
+    flexContainer: [
+      'flexContainer',
+      flexContainerVertical && 'flexContainerVertical',
+      centered && 'centered',
+    ],
+    indicator: ['indicator'],
+    scrollButtons: ['scrollButtons', scrollButtonsHideMobile && 'scrollButtonsHideMobile'],
+  };
+
+  return composeClasses(slots, getTabsUtilityClass, classes);
+};
+
+const TabsRoot = experimentalStyled(
+  'div',
+  {},
+  {
+    name: 'MuiTabs',
+    slot: 'Root',
+    overridesResolver: (props, styles) => {
```

Comment:
What should be the naming convention here because there are many overridesResolver in this file?

Reply:
My point was, we should have `overridesResolver` only for the `Root`, unless some slot is render outside of the root's tree. Otherwise there is no point in adding `overridesResolver` for the slots. So far this was the case only for the `ListItem` and `Select` components.

Reply:
It's outdated, resolving.